src/backends/torch/torchmodule.cc

```diff
@@ -0,0 +1,497 @@
+#include "torchmodule.h"
+
+#include "graph.h"
+#include "native/native.h"
+
+namespace dd
+{
+  // ======= TORCH MODULE
+
+  TorchModule::TorchModule() : _device{ "cpu" }
+  {
+  }
+
+  torch::Tensor TorchModule::to_tensor_safe(const c10::IValue &value)
```

Comment:
This function is not meant to be part of TorchModule, it's just an utility function to safely extract a tensor from an IValue. Put it in another file (`torchutil.h`)?

Reply:
done